docs/src/index.md

```diff
@@ -183,57 +221,65 @@ Similarly every `pullback` returns an extra `∂self`, which for things without
 
 #### Pushforward / Pullback summary
 
-- **Pushforward:**
-    - returned by `frule`
-    - takes input space wiggles, gives output space wobbles
-    - 1 argument per original function argument + 1 for the function itself
-    - 1 return per original function return
-
 - **Pullback**
    - returned by `rrule`
    - takes output space wobbles, gives input space wiggles
    - 1 argument per original function return
    - 1 return per original function argument + 1 for the function itself
 
-#### Pushforward/Pullback and Total Derivative/Gradient
+- **Pushforward:**
+    - part of `frule`
+    - takes input space wiggles, gives output space wobbles
+    - 1 argument per original function argument + 1 for the function itself
+    - 1 return per original function return
+
+
+#### Pullback/Pushforward and Directional Derivative/Gradient
+
+The most trivial use of the `pushforward` from within `frule` is to calculate the directional derivative:
 
-The most trivial use of `frule` and returned `pushforward` is to calculate the [Total Derivative](https://en.wikipedia.org/wiki/Total_derivative):
+If we would like to know the the directional derivative of `f` for an input chage of `(1.5, 0.4, -1)`
 
 ```julia
-y, f_pushforward = frule(f, a, b, c)
-ẏ = f_pushforward(1, 1, 1, 1)  # for appropriate `1`-like perturbation.
+direction = (1.5, 0.4, -1) # (ȧ, ḃ, ċ)
+y, ẏ = frule(f, a, b, c, Zero(), 1, 0, 0)
 ```
 
-Then we have that `ẏ` is the _total derivative_ of `f` at `(a, b, c)`, written mathematically as ``df_{(a,b,c)}``
+On the basis directions one gets the partial derivatives of `y`:
+```julia
+y, dy_da = frule(f, a, b, c, Zero(), 1, 0, 0)
+y, dy_db = frule(f, a, b, c, Zero(), 0, 1, 0)
+y, dy_dc = frule(f, a, b, c, Zero(), 0, 0, 1)
```

Comment:
`dy_da` could also be written `ā`, and `dy_db` could be written `b̄` and so on?

Reply:
The overbar notation is usually reserved for reverse-mode-related quantities, so probably best not to do that here. Might make sense to write these as partials though (to be pedantic):
```julia
∂y_∂a
```
etc